language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -174,57 +191,43 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
         buildProject []
       withCurrentDirectory here $ do
         step "daml2ts..."
-        writePackageJson
-        copyDirectory damlTypes "daml-types"
+        setupYarnEnvironment
         daml2tsProject [groverDar, groverDar] daml2tsDir
-        assertFileExists (groverTsSrc </> "Grover.ts")
-        assertFileExists (groverTsLib </> "Grover.js")
-        assertFileExists (groverTsLib </> "Grover.d.ts")
+        mapM_ (assertTsFileExists groverTs) [ "index", "Grover" ]
 
   , testCaseSteps "DAVL test" $ \step -> withTempDir $ \here -> do
       let daml2tsDir = here </> "daml2ts"
       withCurrentDirectory here $ do
-        copyDirectory damlTypes "daml-types"
         step "daml2ts..."
-        writePackageJson
+        setupYarnEnvironment
         callProcessSilent daml2ts $
           [ davl </> "davl-v4.dar"
           , davl </> "davl-v5.dar"
           , davl </> "davl-upgrade-v4-v5.dar" ] ++
           ["-o", daml2tsDir]
-        assertFileExists (daml2tsDir </> "davl-0.0.4" </> "src" </> "DAVL.ts")
-        assertFileExists (daml2tsDir </> "davl-0.0.4" </> "lib" </> "DAVL.js")
-        assertFileExists (daml2tsDir </> "davl-0.0.4" </> "lib" </> "DAVL.d.ts")
-        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "src" </> "DAVL.ts")
-        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.js")
-        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.d.ts")
-        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
-        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.js")
-        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.d.ts")
+        mapM_ (assertTsFileExists (daml2tsDir </> "davl-0.0.4")) [ "index", "DAVL" ]
+        mapM_ (assertTsFileExists (daml2tsDir </> "davl-0.0.5")) [ "index", "DAVL" ]
```

Comment:
I'm surprised this test actually passes. There is no `DAVL.ts` in v5, there's only `DAVL/V5.ts`. Can you please investigate?

Reply:
Indeed, it's there : https://raw.githubusercontent.com/digital-asset/davl/51d3977be2ab22f7f4434fd4692ca2e17a7cce23/v5/daml/DAVL.daml. I can only suppose that v5 changed in more recent commits?

Reply:
Oh, that explains it. Thanks. Let's move to a newer version of DAVL in another PR.

Reply:
You got it. I shall take care of it.